internal/clusterconfig/recert.go

```diff
@@ -0,0 +1,92 @@
+package clusterconfig
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/clusterinfo"
+	"github.com/openshift-kni/lifecycle-agent/utils"
+)
+
+const recertConfigFile = "recert_config.json"
+
+type recertConfig struct {
+	DryRun            bool     `json:"dry_run,omitempty"`
+	ExtendExpiration  bool     `json:"extend_expiration,omitempty"`
+	EtcdEndpoint      string   `json:"etcd_endpoint,omitempty"`
+	ClusterRename     string   `json:"cluster_rename,omitempty"`
+	SummaryFile       string   `json:"summary_file,omitempty"`
+	StaticDirs        []string `json:"static_dirs,omitempty"`
+	StaticFiles       []string `json:"static_files,omitempty"`
+	CNSanReplaceRules []string `json:"cn_san_replace_rules,omitempty"`
+	UseKey            []string `json:"use_key,omitempty"`
+	UseCert           string   `json:"use_cert,omitempty"`
+}
+
+// CreateRecertConfigFile function to create recert config file
+// those params will be provided to an installation script after reboot
+// that will run recert command with them
+func CreateRecertConfigFile(clusterData *clusterinfo.ClusterInfo,
+	clusterConfigPath string, ostreeDir string) error {
+	// read seed manifest json
+	seedManifestFile := filepath.Join(filepath.Dir(clusterConfigPath), seedManifest)
+	seedClusterInfo := &clusterinfo.ClusterInfo{}
+	err := utils.ReadYamlOrJSONFile(seedManifestFile, seedClusterInfo)
+	if err != nil {
+		return fmt.Errorf("failed to parse %s data, error: %w", seedManifestFile, err)
+	}
+
+	config := recertConfig{
+		DryRun:           false,
+		EtcdEndpoint:     "localhost:2379",
+		ExtendExpiration: true,
+		StaticDirs:       []string{"/kubelet", "/kubernetes", "/machine-config-daemon"},
+		StaticFiles:      []string{"/host-etc/mcs-machine-config-content.json"},
+		ClusterRename:    fmt.Sprintf("%s:%s", clusterData.ClusterName, clusterData.Domain),
+		SummaryFile:      "/var/tmp/recert-summary.yaml",
+	}
+
+	seedFullDomain := fmt.Sprintf("%s.%s", seedClusterInfo.ClusterName, seedClusterInfo.Domain)
+	clusterFullDomain := fmt.Sprintf("%s.%s", clusterData.ClusterName, clusterData.Domain)
+
+	config.CNSanReplaceRules = []string{
+		fmt.Sprintf("%s,%s", seedClusterInfo.MasterIP, clusterData.MasterIP),
+		fmt.Sprintf("api.%s,api.%s", seedFullDomain, clusterFullDomain),
+		fmt.Sprintf("api-int.%s,api-int.%s", seedFullDomain, clusterFullDomain),
+		fmt.Sprintf("*.apps.%s,*.apps.%s", seedFullDomain, clusterFullDomain),
+	}
+
+	if _, err := os.Stat(filepath.Join(ostreeDir, certsDir)); err == nil {
```

Comment:
Shouldn't we have a hard error if certs dir doesn't exist?

Reply:
I took the logic from script, not sure totally what was expected here. Do we always should have it? In case we should always provide it sure we should fail

Reply:
In IBU we should have certs, in IBI we might not need it. But let's figure it out when we're doing IBI

Reply:
cool, let me change this part